cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -98,6 +98,11 @@ public DatastoreManager(String directoryPath) {
      */
     public DatastoreManager(File directoryPath) {
         logger.fine("Datastore path: " + directoryPath);
+
+        if(!directoryPath.exists()){
+            directoryPath.mkdir();
```

Comment:
We don't know for sure, the aim of this PR is to match CDTDatastore behaviour which does the equivalent of `mkdir $path`, so calling `mkdirs()` would then put CDTDatastore out of step with sync-android.

I don't think its worth checking the boolean returned, since we can just use the other checks to ensure that it was created successfully, or can write to it etc.